arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,155 @@
+        return "error", ErrorOutput(
+            f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
+        )
+    return "completed", cmd_out
+
+
+class StartTimerlatStep:
+    exit = Event()
+    finished_early = False
+
+    @plugin.signal_handler(
+        id=predefined_schemas.cancel_signal_schema.id,
+        name=predefined_schemas.cancel_signal_schema.display.name,
+        description=predefined_schemas.cancel_signal_schema.display.description,
+        icon=predefined_schemas.cancel_signal_schema.display.icon,
+    )
+    def cancel_step(self, _input: predefined_schemas.cancelInput):
+        # First, let it know that this is the reason it's exiting.
+        self.finished_early = True
+        # Now signal to exit.
+        self.exit.set()
+
+    @plugin.step_with_signals(
+        id="run-timerlat",
+        name="Run RTLA Timerlat",
+        description=(
+            "Runs the RTLA Timerlat data collection and then processes the results "
+            "into a machine-readable format"
+        ),
+        outputs={"success": TimerlatOutput, "error": ErrorOutput},
+        signal_handler_method_names=["cancel_step"],
+        signal_emitters=[],
+        step_object_constructor=lambda: StartTimerlatStep(),
+    )
+    def run_timerlat(
+        self,
+        params: TimerlatInputParams,
+    ) -> typing.Tuple[str, typing.Union[TimerlatOutput, ErrorOutput]]:
+
+        timerlat_cmd = ["/usr/bin/rtla", "timerlat", "hist"]
+        timerlat_cmd.extend(params.to_flags())
+
+        try:
+            print("Gathering data... Use Ctrl-C to stop.")
+            proc = subprocess.Popen(
+                timerlat_cmd,
+                start_new_session=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+
+            # Block here, waiting on the cancel signal
+            self.exit.wait(params.duration)
+
+        except subprocess.CalledProcessError as err:
+            return "error", ErrorOutput(
+                f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
+            )
+
+        # Secondary block interrupt is via the KeyboardInterrupt exception.
+        # This enables running the plugin stand-alone without a workflow.
+        except (KeyboardInterrupt, SystemExit):
+            print("\nReceived keyboard interrupt; Stopping data collection.\n")
+            self.finished_early = True
+
+        # In either the case of a keyboard interrupt or a cancel signal, we need to
+        # send the SIGINT to the subprocess.
+        if self.finished_early:
+            proc.send_signal(2)
+
+        output, _ = proc.communicate()
+
+        latency_hist = []
+        total_irq_latency = {}
+        total_thr_latency = {}
+        total_usr_latency = {}
+        stats_names = ["over:", "count:", "min:", "avg:", "max:"]
+        stats_per_col = []
+        found_all = False
+
+        for line in output.splitlines():
```

Comment:
This section could use some comments. I have little idea on what it's doing since I'm not familiar with the output format.